include/WCSimNuHepMC3Reader.hh

```diff
@@ -0,0 +1,64 @@
+#ifndef WCSimNuHepMC3Reader_h
+#define WCSimNuHepMC3Reader_h
+
+#include <G4LorentzVector.hh>
+#include <G4String.hh>
+#include <memory>
+
+// #include "HepMC3/GenParticle.h"
+#include "HepMC3/FourVector.h"
+#include "HepMC3/GenEvent.h"
+#include "HepMC3/ReaderAscii.h"
+#include "WCSimDetectorConstruction.hh"
+
+namespace HepMC3 {
+class Reader;
+typedef std::shared_ptr<Reader> ReaderPtr;
+} // namespace HepMC3
+
+class WCSimNuHepMC3Reader {
+  public:
+    WCSimNuHepMC3Reader(std::string filename, WCSimDetectorConstruction *myDC);
+    ~WCSimNuHepMC3Reader();
+
+    // Read the next event from the input file
+    bool ReadEvent(bool positionFlag);
+
+    // Vector of particles in event
+    std::vector<HepMC3::GenParticlePtr> particles;
+
+    // Populate particles using event.particles()
+    bool GetParticles();
+
+    // Return Lorentz vector of a specific particle
+    G4LorentzVector GetLorentzVector(int particle_index);
+
+    // Generate a random position in the detector construiction
+    HepMC3::FourVector GenRandomPosition();
+
+    // Generate a random direction
+    HepMC3::FourVector GenRandomDirection();
+
+    // Rotate the event
+    void RotateEvent();
+
```

Comment:
```suggestion
```
Not defined (it suprised me that it doesn't complain about a missing symbol when running...)